File: examples/training/01-intro/step2-with-logging.py
from agentuity import AgentRequest, AgentResponse, AgentContext
import json
from datetime import datetime

async def run(request: AgentRequest, response: AgentResponse, context: AgentContext):
    context.logger.info('Agent invoked', {'contentType': request.data.content_type})

    received_data = None

    if request.data.content_type == 'application/json':
        received_data = await request.data.json()
    elif request.data.content_type == 'text/plain':
        received_data = await request.data.text()
    else:
        received_data = await request.data.text()

    context.logger.info('Request received', {'data': received_data})

    return response.json({
        'message': 'Request received and logged',
        'contentType': request.data.content_type,
        'received': received_data,
        'timestamp': datetime.utcnow().isoformat() + 'Z'
    })

def welcome():
    return {
        'welcome': 'Send data and watch the logs appear in DevMode.',
        'prompts': [
            {'data': json.dumps({'action': 'test', 'value': 123}), 'contentType': 'application/json'},
            {'data': 'Testing logs', 'contentType': 'text/plain'}
        ]
    }

File: examples/training/01-intro/step2-with-logging.ts
import type { AgentRequest, AgentResponse, AgentContext } from "@agentuity/sdk";

export default async function Agent(
  request: AgentRequest,
  response: AgentResponse,
  context: AgentContext
) {
  context.logger.info('Agent invoked', { contentType: request.data.contentType });

  let receivedData: unknown = null;

  switch (request.data.contentType) {
    case 'application/json':
      receivedData = await request.data.json();
      break;
    case 'text/plain':
      receivedData = await request.data.text();
      break;
    default:
      receivedData = await request.data.text();
  }

  context.logger.info('Request received', { data: receivedData });

  return response.json({
    message: 'Request received and logged',
    contentType: request.data.contentType,
    received: receivedData,
    timestamp: new Date().toISOString()
  });
}

export const welcome = () => ({
  welcome: 'Send data and watch the logs appear in DevMode.',
  prompts: [
    { data: JSON.stringify({ action: 'test', value: 123 }), contentType: 'application/json' },
    { data: 'Testing logs', contentType: 'text/plain' }
  ]
});
